Monika After Story/game/script-ch30.rpy

```diff
@@ -1778,14 +1778,21 @@ label ch30_reset:
         #TODO: Fix the game unlock/is unlocked system to account for conditions like these
         #mas_isGameUnlocked should NOT return True if we're failing this condition because we use it elsewhere
         game_unlock_db = {
-            "pong": "ch30_main", # pong should always be unlocked
             "chess": "mas_unlock_chess",
             mas_games.HANGMAN_NAME: "mas_unlock_hangman",
             "piano": "mas_unlock_piano",
         }
+        mas_unlockGame("pong") # always unlock pong
 
         for game_name, game_startlabel in game_unlock_db.iteritems():
-            if not mas_isGameUnlocked(game_name) and renpy.seen_label(game_startlabel):
+            # try grabbing event if possible
+            game_unlock_ev = mas_getEV(game_startlabel)
+            if game_unlock_ev:
+                if game_unlock_ev.shown_count > 0:
+                    mas_unlockGame(game_name)
+
+            elif renpy.seen_label(game_startlabel):
+                # otherwise, just use label check
```

Comment:
that is intended. We don't know if you have seen the event so we can't determine if we should unlock it or not.

Reply:
But in this case there are 2 problems. The first is that the ev exists so we can never reach the elif and we can't use the label check. The second is that on very old persists there are cases where topics, like piano and hangman, have a `shown_count` of 0 but the label is seen and as the label is seen it has no conditionals so the game can't be unlocked, so in cases like these we can't rely purely on `shown_count`

Reply:
the elif is a fallback in case the event fails to be created or found for whatever reason. The event is still the source of truth and should be used as the primary and only check if we do find it. However I did see that if a user from 0.7.1 and prior did not update until after 0.10.0, they would end up with no game unlocks, so I'll accept changing this line *and* add an update script to not deal with this uncertainty anymore.